cnd/src/network/comit_ln.rs

```diff
@@ -1,49 +1,63 @@
 use crate::{
-    asset,
-    btsieve::ethereum::{Cache, Web3Connector},
-    htlc_location, identity,
+    asset, identity,
     network::{
         oneshot_behaviour,
         protocols::{
             announce,
-            announce::{
-                behaviour::{Announce, BehaviourOutEvent},
-                SwapDigest,
-            },
+            announce::{behaviour::Announce, SwapDigest},
             ethereum_identity, finalize, lightning_identity, secret_hash,
         },
     },
     seed::{DeriveSwapSeedFromNodeLocal, RootSeed},
     swap_protocols::{
-        halight::{self, LndConnectorAsReceiver, LndConnectorAsSender, LndConnectorParams},
-        han, herc20, ledger,
+        halight,
+        halight::States,
         ledger::{ethereum::ChainId, lightning, Ethereum},
         rfc003::{create_swap::HtlcParams, DeriveSecret, Secret, SecretHash},
-        state::Update,
         CreateSwapParams, LedgerStates, NodeLocalSwapId, Role, SwapId,
     },
     timestamp::Timestamp,
-    transaction,
 };
 use blockchain_contracts::ethereum::rfc003::ether_htlc::EtherHtlc;
-use chrono::Utc;
 use digest::Digest;
-use futures::{AsyncWriteExt, TryStreamExt};
+use futures::AsyncWriteExt;
 use libp2p::{
-    swarm::{NetworkBehaviour, NetworkBehaviourEventProcess},
+    swarm::{
+        NetworkBehaviour, NetworkBehaviourAction, NetworkBehaviourEventProcess, PollParameters,
+    },
     NetworkBehaviour,
 };
-use std::{collections::HashMap, sync::Arc};
-use tracing_futures::Instrument;
+use std::{
+    collections::{HashMap, VecDeque},
+    sync::Arc,
+    task::{Context, Poll},
+};
+
+/// Event emitted  by the `ComitLn` behaviour.
+#[derive(Debug)]
+pub enum BehaviourOutEvent {
+    SwapIsFinalizedCanStartExecution {
+        local_swap_id: NodeLocalSwapId,
+        swap_params: CreateSwapParams,
+        secret_hash: SecretHash,
+        invoice_states: Arc<States>,
```

Comment:
I don't think these should be part of the event, they are global dependencies so they should go into `ComitNode` as fields.

Reply:
oh, true, I missed that those are actually just used for execution. Will more them.